doc/src/modules/ntheory.rst

```diff
@@ -192,3 +192,46 @@ Ntheory Functions Reference
 .. module:: sympy.ntheory.bbp_pi
 
 .. autofunction:: pi_hex_digits
+
+.. module:: sympy.ntheory.ecm
+
+.. autofunction:: ecm
+
+The ecm function is a subexponential factoring algorithm capable of factoring
+numbers of around ~35 digits comfortably within few seconds. The time complexity
+of ecm is dependent on the smalled proper factor of the number. So even if the
+number is really large but its factors are comparatively smaller then ecm will
+can easily factor them. For example we take N with 15 digit factors
+{15154262241479, 15423094826093, 799333555511111, 809709509409109, 888888877777777,
+914148152112161}. Now N is a 87 digit number. Ecm takes under 30secs to factorize
```

Comment:
In the example provided to justify this statement, I think it would be better if we show the timing results using `timeit`.